pkg/v18/resource/clusterconfigmap/desired.go

```diff
@@ -20,8 +20,14 @@ func (r *StateGetter) GetDesiredState(ctx context.Context, obj interface{}) ([]*
 
 	configMapName := key.ClusterConfigMapName(clusterConfig)
 
+	clusterDNSIP, err := key.DNSIP(r.clusterIPRange)
```

Comment:
Please add a comment here that we calculate the cluster DNS IP from the IP range. I'd forgotten this was how we do it and it could be useful for future us.